core/src/test/scala/codesearch/core/regex/space/SpaceInsensitiveSpec.scala

```diff
@@ -0,0 +1,49 @@
+package codesearch.core.regex.space
+
+import org.scalatest.{FreeSpec, Matchers}
+
+class SpaceInsensitiveSpec extends FreeSpec with Matchers {
+  def addingSpaceInsensitive(before: String, after: String) = {
+    before in {
+      SpaceInsensitive.spaceInsensitiveString(before) shouldBe after
+    }
+  }
+
+  "Space Insensitive tests" - {
+    "Simple strings" - {
+      addingSpaceInsensitive("Hello world", "Hello +world")
+
+      addingSpaceInsensitive("Hello    world", "Hello    world")
+
+      addingSpaceInsensitive("foo\\ bar", "foo\\ bar")
+    }
+
+    "Strings with regexp" - {
+      addingSpaceInsensitive("foo +bar", "foo +bar")
+
+      addingSpaceInsensitive("foo ?bar", "foo( +)?bar")
+
+      addingSpaceInsensitive("foo *bar", "foo *bar")
+    }
+
+    "Strings with other cases" - {
+      addingSpaceInsensitive("foo [bar]", "foo +[bar]")
+
+      addingSpaceInsensitive("foo {1,3}", "foo {1,3}")
+
+      addingSpaceInsensitive("foo [b ar]", "foo +[b ar]")
+    }
+
+    "*Not valid tests*" - {
+      addingSpaceInsensitive("foo {bar}", "foo {bar}")
```

Comment:
You should also leave a comment saying why the tests are not valid, so that if somebody else works on this code, they wouldn't have to scratch their head.

You can just copypaste my comment, for instance.